meshnets/utils/data_processing.py

```diff
@@ -138,3 +104,75 @@ def mesh_file_to_graph_data(file_path: str,
             logging.info('Pressure label shape : %s', graph.y.shape)
 
     return graph
+
+
+def make_node_features(file_path: str,
+                       wind_vector: Tuple[float],
+                       verbose: bool = False) -> np.ndarray:
+    """Receive the path to a mesh file (e.g. `.obj`, `.vtk` file) and
+    a wind vector.
+
+    Args:
+        file_path: Path to the mesh file
+        wind_vector: Wind vector
+        verbose: Whether to print information about the mesh file
+    Returns:
+        node_features: Node features for each node of the mesh
+    """
+    if verbose:
+        logging.info('Loading mesh data from %s', file_path)
+    nodes, _, _ = data_loading.load_edge_mesh_pv(file_path,
+                                                 load_pressure=False,
+                                                 verbose=verbose)
+
+    # node features for each node are the wind vector
+    node_features = np.tile(wind_vector, (len(nodes), 1))
+    return node_features
+
+
+def make_edge_index_and_features(file_path: str,
+                                 verbose: bool = False
+                                ) -> Tuple[np.ndarray, np.ndarray]:
+    """Receive the path to a mesh file (e.g. `.obj`, `.vtk` file) and
+    a wind vector. Return the edge index and the edge features.
+
+    Args:
+        file_path: Path to the mesh file
+        verbose: Whether to print information about the mesh file
+    Returns:
+        edge_index: Edge index for the mesh
+        edge_attr: Edge features for the mesh
+    """
+    nodes, edges, _ = data_loading.load_edge_mesh_pv(file_path,
```

Comment:
seems like each of these 3 functions is loading the data from the file, and then discarding some of its info. Can we load it just once (avoid code redundancy AND make it faster) and then pass around the relevant data to this functions?

Reply:
This a good point. Will change it on my next commit